### ISSUE TITLE
select2_ajax Filter issue

### ISSUE THREAD

**Issue by Athuli7** [Laravel-Backpack/CRUD#2102]
# Bug report

### What I did

Upgraded to Backpack 4

### What I expected to happen

Everything to work after the upgrade guide.

### What happened

Most of it worked, but select2_ajax filter was broken. Ie, if you use more than 1 in the same CRUD, then all filters will use the url of the last select2_ajax filter.

### What I've already tried to fix it

Fix the jQuery selector.

### Backpack, Laravel, PHP, DB version

When I run ```php artisan backpack:version``` the output is:
```
### PHP VERSION:
PHP 7.3.8 (cli) (built: Aug  6 2019 23:15:07) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.3.8, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.3.8, Copyright (c) 1999-2018, by Zend Technologies

### LARAVEL VERSION:
laravel/framework                            v6.0.4            

### BACKPACK VERSION:
backpack/crud                                4.0.5             
backpack/generators                          2.0.4             
backpack/permissionmanager                   5.0.0             
backpack/settings                            3.0.0             
```

### Current version of [select2_ajax.blade.php](https://github.com/Laravel-Backpack/CRUD/blob/master/src/resources/views/crud/filters/select2_ajax.blade.php)

```
    <script>
        jQuery(document).ready(function($) {
            // trigger select2 for each untriggered select2 box
            $('[data-filter-type=select2_ajax]').each(function () {
            	var filterName = $(this).attr('data-filter-name');
```

### Modified version that works is
```
    <script>
        jQuery(document).ready(function($) {
            // trigger select2 for each untriggered select2 box
            $('#filter_{{ $filter->name }}').each(function () {
            	var filterName = $(this).attr('data-filter-name');
```

I'm not sure why there is an each loop. Can someone help me out by saying why?

**Comment by martijnb92** [comment]
Most likely the same as #2075

**Comment by Athuli7** [comment]
Setting `persistentTable` to `false` helps but at the cost of that feature.

**Comment by martijnb92** [comment]
> Setting `persistentTable` to `false` helps but at the cost of that feature.

Unfortunately that does not help

**Comment by tabacitu** [comment]
You're totally right @Athuli7 . Thank you for raising this, and providing the solution. I've just merged it into master, and it'll be available with a ```composer update``` later today, as 4.0.12.

Honestly I don't remember why we moved away from IDs in jQuery selectors. I do remember why we did that in fields, but in filters... I don't. Maybe for consistency across fields-filters, or something like that. But I can confirm it did NOT work before your change, and it DOES work after your change. So let's go that way :-)

Thanks again.
Cheers!
